friends/src/App.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { Route, Redirect } from "react-router-dom";
+import { FriendsList } from "./components/FriendsList";
+import { NewFriendForm } from "./components/NewFriendForm";
+import "./App.css";
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: []
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get("http://localhost:5000/friends")
+      .then(res => {
+        this.setState({ friends: res.data });
+      })
+      .catch(err => {
+        this.setState({ error: err });
+      });
+  }
+
+  addNewFriend = () => {};
+
+  render() {
+    return (
+      <div className="App">
+        <Route
+          exact
+          path="/"
+          component={() => <Redirect to="/friendslist" />}
+        />
+        <Route
+          exact
+          path="/friendslist"
+          render={routeProps => (
+            <FriendsList {...routeProps} friends={this.state.friends} />
```

Comment:
I'm glad to see you understand how Redirects work, but I'd still argue against the necessity of it here: why not just have the FriendList component render at the "/" path and take out your redirect?